frontend/shared/types.hpp

```diff
@@ -1,3 +1,12 @@
+/**
+ * @file types.hpp
+ * @brief Defines some datatypes and the fold and unfold function for each.
+ *
+ * - fold: compress datatype to writer, returns size of compressed object
```

Comment:
I would not call folding "compress", it has really different semantics.
we fold because we compare keys in lexicographical order, the tree does not understand the semantic, it just compare raw bytes.
Moreover, I don't like writing comments this way.
I prefer compact comments above the line
```
// Something
```
https://github.com/adnan-alhomssi/leanstore/commit/a5515c96df6221e7fd23e95ae0a7afad51561d6f